sdk/src/main/java/io/horizen/utxo/box/CrossChainMessageBox.java

```diff
@@ -0,0 +1,49 @@
+package io.horizen.utxo.box;
+
+import io.horizen.proposition.PublicKey25519Proposition;
+import io.horizen.sc2sc.CrossChainProtocolVersion;
+import io.horizen.utxo.box.data.CrossChainMessageBoxData;
+
+import static io.horizen.utxo.box.CoreBoxesIdsEnum.CrossChainMessageBoxId;
+
+public final class CrossChainMessageBox
```

Comment:
During the review of sc2sc `CrossChainMessage` part and `CrossChainMessageBox` I saw the following ideas:
1. `CrossChainMessage ` is an interface with one particular implementation `CrossChainMessageImpl`. All the validation code was relying on that interface. So, theoretically application developers may define their own implementations. I guess the main reason was to have a contextual `payload` type (not raw bytes) in the ancestor if needed.
2. `CrossChainMessageBox` is the final class. So, we force developers to use it if they want to have sc2sc. `payload` is always raw (bytes), no context from the core perspective. But on application side developer must know how to decode this raw data to something that have sense -> then to validate it according to the application rules.

In the comments for `CrossChainMessage` I put some arguments of making it as a `public final class` instead of an `interface`.

Now I'm thinking about the benefits of having `CrossChainMessageBox` as `public final`. We may keep it as is, but it seems that we should do an opposite:

1. Make it as a `public abstract`. The same for `CrossChainMessageBoxData`
2. Add a `final` methods implementation of `getCrossChainMessage` and several others.
3. In all the core validation rules deal with these abstract classes without additional context, but allow developers to extend them and define the exact payload data structure.
4. Keep undefined the boxId of `CrossChainMessageBox` and remove `CrossChainMessageBox` from core boxes and core box serializes.

The very same logic must be applied to `CrossChainRedeemMessageBox`